workitem.2.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+	query "github.com/almighty/almighty-core/query/simple"
+	"github.com/almighty/almighty-core/transaction"
+	"github.com/goadesign/goa"
+)
+
+// Workitem2Controller implements the workitem.2 resource.
+type Workitem2Controller struct {
+	*goa.Controller
+	wiRepository models.WorkItemRepository
+	ts           transaction.Support
+}
+
+// NewWorkitem2Controller creates a workitem.2 controller.
+func NewWorkitem2Controller(service *goa.Service, wiRepository models.WorkItemRepository, ts transaction.Support) *Workitem2Controller {
+	return &Workitem2Controller{Controller: service.NewController("WorkitemController"), wiRepository: wiRepository, ts: ts}
+}
+
+// List runs the list action.
+// Prev and Next links will be present only when there actually IS a next or previous page.
+// Last will always be present. Total Item count needs to be computed from the "Last" link.
+func (c *Workitem2Controller) List(ctx *app.ListWorkitem2Context) error {
+	// Workitem2Controller_List: start_implement
+
+	exp, err := query.Parse(ctx.Filter)
+	if err != nil {
+		return goa.ErrBadRequest(fmt.Sprintf("could not parse filter: %s", err.Error()))
+	}
+	start, limit, err := parseLimit(ctx.Page)
+	if err != nil {
+		return goa.ErrBadRequest(fmt.Sprintf("could not parse paging: %s", err.Error()))
+	}
+	return transaction.Do(c.ts, func() error {
```

Comment:
Just to make a note; The execution overhead is minimal atm, but...

* The more code you add within the TX block the longer you're keeping the TX open and resources potentially locked in DB (A TX which in this case is not needed at all)
* The more code you add the more reasons you give the TX to end up in Rollback due to non db related errors.